### ISSUE TITLE
Get spurious error about refactoring a file that does not support "refactoring"

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#41532]
**Version Used**: 
Version 16.6.0 Preview 1.0 [29809.2.master]

**Steps to Reproduce**:

1. Inline rename line indicated below.

``` C#
// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.

using System;

namespace Microsoft.VisualStudio.ProjectSystem.OperationProgress
{
    internal class ProgressTrackerOutputDataSource : IProgressTrackerOutputDataSource, IJoinableProjectValueDataSource
    {
        public ProgressTrackerOutputDataSource(IJoinableProjectValueDataSource underlyingDataSource, ConfiguredProject configuredProject, string operationProgressStageId, string name, string displayMessage)
        {
            DataSource = underlyingDataSource; // < Rename "DataSource"
            ConfiguredProject = configuredProject;
            OperationProgressStageId = operationProgressStageId;
            Name = name;
            DisplayMessage = displayMessage;
        }

        public IJoinableProjectValueDataSource DataSource { get; }

        public ConfiguredProject ConfiguredProject { get; }

        public string OperationProgressStageId { get; }

        public string Name { get; }

        public string DisplayMessage { get; }

        public IDisposable? Join()
        {
            return DataSource.Join();
        }
    }
}

```
**Expected Behavior**:
No error

**Actual Behavior**:
![image](https://user-images.githubusercontent.com/1103906/74120627-1f85a600-4c18-11ea-9233-86f7acac2010.png)

This property is not consumed by anything else other than in this file.

**Comment by ryzngard** [comment]
@genlu does this look like the area you fixed?

**Comment by genlu** [comment]
The change I made simply ignore unchangeable documents and proceed, not sure where this dialog comes from, which I think is actually a better experience than the fix I made (if it shows which file is unsupported)

**Comment by CyrusNajmabadi** [comment]
I could not find taht string in roslyn.  Is this actually in our code, or something in the editor?

**Comment by davkean** [comment]
Hmm, that's coming from CPS, let me see if I can repro this.

**Comment by davkean** [comment]
This is a XAML bug, let me move it:

```
>	mscorlib.dll!System.Threading.Tasks.Task.ThrowIfExceptional(bool includeTaskCanceledExceptions) Line 2153	C#
 	mscorlib.dll!System.Threading.Tasks.Task<Microsoft.VisualStudio.DesignTools.Xaml.LanguageService.Semantics.IXamlFileInformationContext>.GetResultCore(bool waitCompletionNotification) Line 556	C#
 	mscorlib.dll!System.Threading.Tasks.Task<System.__Canon>.Result.get() Line 532	C#
 	[Waiting on Async Operation, double-click or press enter to view Async Call Stacks]	
 	Microsoft.VisualStudio.DesignTools.XamlLanguageService.dll!Microsoft.VisualStudio.DesignTools.Xaml.LanguageService.XamlLanguageService.GetFileContext(string fileName, Microsoft.VisualStudio.TextManager.Interop.IVsTextLines textLines) Line 798	C#
 	Microsoft.VisualStudio.DesignTools.XamlLanguageService.dll!Microsoft.VisualStudio.DesignTools.Xaml.LanguageService.XamlLanguageService.CreateContextIfReferences(string[] rglpszRQNames, Microsoft.VisualStudio.DesignTools.Xaml.LanguageService.Semantics.ReferenceOptions options, Microsoft.VisualStudio.DesignTools.Xaml.LanguageService.XamlLanguageService.FileContext fileContext, out Microsoft.VisualStudio.Shell.Interop.IVsInvisibleEditor invisibleEditor) Line 1889	C#
 	Microsoft.VisualStudio.DesignTools.XamlLanguageService.dll!Microsoft.VisualStudio.DesignTools.Xaml.LanguageService.XamlLanguageService.Microsoft.VisualStudio.Shell.Interop.IVsRefactorNotify.OnBeforeGlobalSymbolRenamed(Microsoft.VisualStudio.Shell.Interop.IVsHierarchy hierarchy, uint itemid, uint cRQNames, string[] rglpszRQName, string lpszNewName, out System.Array prgAdditionalCheckoutVSITEMIDs) Line 2208	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.Microsoft.VisualStudio.Shell.Interop.IVsHierarchyRefactorNotify.OnBeforeGlobalSymbolRenamed.AnonymousMethod__0(Microsoft.VisualStudio.Shell.Interop.IVsRefactorNotify notify, uint itemID, System.Collections.Generic.HashSet<string> additionalFiles) Line 43	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.HandleBeforeRefactorNotify.AnonymousMethod__0() Line 170	C#
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.ExecuteJob<Microsoft.VisualStudio.ProjectSystem.VS.HResult>(System.Func<System.Threading.Tasks.Task> asyncMethod, Microsoft.VisualStudio.Threading.JoinableTask job) Line 699	C#
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.RunAsync<Microsoft.VisualStudio.ProjectSystem.VS.HResult>(System.Func<System.Threading.Tasks.Task<Microsoft.VisualStudio.ProjectSystem.VS.HResult>> asyncMethod, bool synchronouslyBlocking, Microsoft.VisualStudio.Threading.JoinableTaskCreationOptions creationOptions) Line 688	C#
 	Microsoft.VisualStudio.Threading.dll!Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run<Microsoft.VisualStudio.ProjectSystem.VS.HResult>(System.Func<System.Threading.Tasks.Task<Microsoft.VisualStudio.ProjectSystem.VS.HResult>> asyncMethod, Microsoft.VisualStudio.Threading.JoinableTaskCreationOptions creationOptions) Line 503	C#
 	Microsoft.VisualStudio.ProjectSystem.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.ProjectMultiThreadedService.ExecuteSynchronously<Microsoft.VisualStudio.ProjectSystem.VS.HResult>(System.Func<System.Threading.Tasks.Task<Microsoft.VisualStudio.ProjectSystem.VS.HResult>> asyncAction) Line 133	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.HrInvoke.AnonymousMethod__0() Line 3633	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.dll!Microsoft.VisualStudio.ProjectSystem.VS.HResult.Invoke(System.Func<Microsoft.VisualStudio.ProjectSystem.VS.HResult> action, System.IServiceProvider vsShellServiceProvider, Microsoft.VisualStudio.ProjectSystem.IProjectFaultHandlerService projectFaultHandlerService, Microsoft.VisualStudio.ProjectSystem.UnconfiguredProject project) Line 320	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.HrInvoke(System.Func<System.Threading.Tasks.Task<Microsoft.VisualStudio.ProjectSystem.VS.HResult>> asyncAction, bool registerProjectFaultHandlerService)	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.HandleBeforeRefactorNotify(uint cItemsAffected, uint[] rgItemsAffected, int promptContinueOnFail, System.Func<Microsoft.VisualStudio.Shell.Interop.IVsRefactorNotify, uint, System.Collections.Generic.HashSet<string>, Microsoft.VisualStudio.ProjectSystem.VS.HResult> handler) Line 125	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode.Microsoft.VisualStudio.Shell.Interop.IVsHierarchyRefactorNotify.OnBeforeGlobalSymbolRenamed(uint cItemsAffected, uint[] rgItemsAffected, uint cRQNames, string[] rglpszRQName, string lpszNewName, int promptContinueOnFail) Line 41	C#
 	Microsoft.VisualStudio.ProjectSystem.VS.Implementation.dll!Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNodeWrapper.OnBeforeGlobalSymbolRenamed(uint cItemsAffected, uint[] rgItemsAffected, uint cRQNames, string[] rglpszRQName, string lpszNewName, int promptContinueOnFail) Line 749	C#
 	[Native to Managed Transition]	

```

**Comment by davkean** [comment]
Closed in lieu of https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1094630.
